### PR TITLE
feat: migration to nextjs

### LINKED ISSUE #1470: [FEATURE] Migrate to Next.js
### Description

By moving to next.js we could add and use several features by benefiting from the server component: 

1. Use SSR to prerender our pages on the server for performance optimization
1. Maybe use [ISR](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration) to dynamically re-render the index page to highlight a different profile every day on the index page
1. Use a database connection to store how often a page got clicked
1. Don't use PrimeReact, but use Tailwind for creating our custom components

wdyt @eddiejaoude 

Things like adding Cypress and/or other addition could come in future PRs. I think for us it's important to migrate with an MVP first.

We could use a `staging` branch, similar on how we did it for the API rewrite, to have things a bit more atomic, and not just a single big-bang

## TO DO

- [ ] by default hide stats and enable with json config

### PR DESCRIPTION
**Pull request by eddiejaoude**
fixes #1470 

deployed beta version https://beta.linkfree.eddiehub.io/

## Must have features
- [x] Load profile data
- [x] Stats for page views and link clicks (check `@TODO` in codebase)
- [ ] Storybook components https://github.com/EddieHubCommunity/LinkFree/issues/1944
   - [ ] update build to GitHub Action (remove cypress, add Playwright)
- [ ] Automated tests https://github.com/EddieHubCommunity/LinkFree/issues/1920
   - [x] add to GitHub Action
- [ ] Documentation (quickstart for: local project setup, tests, storybook)
- [x] Add homepage mockup 
- [x] Deploy to something like beta.linkfree.eddiehub.io
- [x] Click link in profile to visit link (currently marked as `TODO` in code)
- [x] Icons for links
- [x] Colours for links

## Deployment
- [x] Migrate all `data` files over
- [x] Removed Storybook dependencies as conflicting with NextJS 13 (need to add back later)
- [x] Dockerfile using `npm install` - need to change to `npm ci` later

## Future Ideas
- [ ] ~~Flag to redirect people to beta version for their profile~~
- [ ] Icons section for socials
- [ ] Profiles to have daily views/clicks stored
- [ ] Graphs for homepage
- [ ] OAuth login (not to use json file) - manage data with form
   - [ ] save list of favourite profiles
- [ ] SEO optimisation
- [ ] Accessibility improvements
- [ ] Multiple themes for the profile page
- [ ] Docs inside the app
- [ ] Events to have click stats
- [ ] 6 profiles to homepage

Screenshot 2022-08-05 at 12 07 19

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/1471"></a>

Screenshot 2022-09-08 at 23 52 43

### CODE REVIEW

**Comment by scottjwilson** [comment]
Hello, I'd like to help out with this. I searched the build logs and I added some missing key props. 
Screen Shot 2022-07-24 at 11 55 46 AM

Not sure if you guys like how I did it though

**Comment by eddiejaoude** [comment]
hey @scottjwilson , thank you for the suggestion! It looks like an improvement, so feel free to create a PR to this open PR 👍

File: README.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+
+## QUICKSTART
+
+1. copy the `.env.example` file to `.env` and update any details required
+2. `npm install`
+3. `npm run dev`
+
+
+
+
```
Comment (schmelto) [inline README.md:12]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
 [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree) | ![Uptime](https://img.shields.io/endpoint?url=https%3A%2F%2Fraw.githubusercontent.com%2FEddieHubCommunity%2Fmonitoring%2Fmaster%2Fapi%2Flink-free%2Fuptime.json)
 
+
```
Comment (schmelto) [inline README.md:3]:
```suggestion
```

**Review by schmelto — APPROVED** [review]

**Comment by Pierstoval** [comment]
Hello! Question about this quite-unreviewable PR (thanks Github for the impossibility to review huge diffs 😅): how much is it cross-compatible with existing hosted apps? Like if I have my `public/data/myusername.json` file, is the new next.js-based app working with the same file?

**Comment by eddiejaoude** [comment]
@Pierstoval this PR has been reviewed as we went along, by merging other smaller PRs into this one - so it should be ok

The previous data files are fine, they only changed location, but before deploying this (very soon), we will make sure any new/updated profiles are moved to the correct location

**Comment by Pierstoval** [comment]
Thanks! I have a "customized" version of the app right now that I self-host, and I can't wait for the next version to be published so that static site generation is *truly* static site and not just a JS-only SPA 😅

**Comment by eddiejaoude** [comment]
v2 also have statistics as well, so you will need a mongo DB...  

Screenshot 2022-11-03 at 11 22 37

**Comment by Pierstoval** [comment]
Will check, I can still change the static file server into a reverse proxy to a nodejs server, that's fine if  it's for analytics :)

**Comment by schmelto** [comment]
Markdown bio is not working on search and popular profiles

**Comment by eddiejaoude** [comment]
Thanks @schmelto !! Fixed here https://github.com/EddieHubCommunity/LinkFree/pull/1471/commits/45bc4d076ce3a732f66cf9f700b57def91791b2e

**Review by Pradumnasaraf — APPROVED** [review]
Looks good to merge. Thank you for your contribution! :tada:

**Review by RakeshSangem — APPROVED** [review]

**Pull request merged**